packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -351,13 +384,99 @@ contract VotingV2 is
-            requestSlashingTrackers.push(
-                SlashingTracker(wrongVoteSlashPerToken, noVoteSlashPerToken, totalSlashed, totalCorrectVotes)
+
+            requestSlashingTrackers[i] = SlashingTracker(
+                wrongVoteSlashPerToken,
+                noVoteSlashPerToken,
+                totalSlashed,
+                totalCorrectVotes
             );
+
             lastRequestIndexConsidered = i + 1;
         }
     }
 
+    /****************************************
+     *       SPAM DELETION FUNCTIONS        *
+     ****************************************/
+
+    function signalRequestsAsSpamForDeletion(uint256[2][] memory spamRequestIndices) public {
+        votingToken.transferFrom(msg.sender, address(this), spamDeletionProposalBond);
+        uint256 currentTime = getCurrentTime();
+        uint256 currentRoundId = voteTiming.computeCurrentRoundId(currentTime);
+        uint256 runningValidationIndex;
+        for (uint256 i = 0; i < spamRequestIndices.length; i++) {
+            // Check request end index is greater than start index.
+            require(spamRequestIndices[i][0] <= spamRequestIndices[i][1], "Bad start index");
+
+            // check the endIndex is less than the total number of requests.
+            require(spamRequestIndices[i][1] < priceRequestIds.length, "Bad end index");
+
+            // Validate index continuity. This checks that each sequential element within the spamRequestIndices
+            // array is sequently and increasing in size.
+            require(spamRequestIndices[i][1] > runningValidationIndex, "Bad index continuity");
+            runningValidationIndex = spamRequestIndices[i][1];
+        }
+        // todo: consider if we want to check if the most recent price request has been settled?
+
+        spamDeletionProposals.push(SpamDeletionRequest(spamRequestIndices, currentTime, false, msg.sender));
+        uint256 proposalId = spamDeletionProposals.length - 1;
+
+        bytes32 identifier = SpamGuardIdentifierLib._constructIdentifier(proposalId);
+
+        _requestPrice(identifier, currentTime, "", true);
+    }
+
+    function executeSpamDeletion(uint256 proposalId) public {
+        require(spamDeletionProposals[proposalId].executed == false, "Already executed");
+        spamDeletionProposals[proposalId].executed = true;
+        bytes32 identifier = SpamGuardIdentifierLib._constructIdentifier(proposalId);
+
+        (bool hasPrice, int256 resolutionPrice, ) =
+            _getPriceOrError(identifier, spamDeletionProposals[proposalId].requestTime, "");
+        require(hasPrice, "Price not yet resolved");
+
+        // If the price is 1e18 then the spam deletion request was correctly voted on to delete the requests.
+        if (resolutionPrice == 1e18) {
+            // Delete the price requests associated with the spam.
+            for (uint256 i = 0; i < spamDeletionProposals[proposalId].spamRequestIndices.length; i++) {
```

Comment:
If there's a _lot_ of spam, would this method break because it's O(n) in spam requests?